pkg/user/api/validation/validation.go

```diff
@@ -27,18 +57,76 @@ func ValidateUserName(name string, prefix bool) (bool, string) {
 func ValidateUser(user *api.User) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
 	allErrs = append(allErrs, kvalidation.ValidateObjectMeta(&user.ObjectMeta, false, ValidateUserName).Prefix("metadata")...)
+	for index, identity := range user.Identities {
+		if ok, msg := ValidateIdentityName(identity, false); !ok {
+			allErrs = append(allErrs, errs.NewFieldInvalid(fmt.Sprintf("identities[%d]", index), identity, msg))
+		}
+	}
+	return allErrs
+}
+
+func ValidateUserUpdate(user *api.User, old *api.User) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	allErrs = append(allErrs, kvalidation.ValidateObjectMetaUpdate(&old.ObjectMeta, &user.ObjectMeta).Prefix("metadata")...)
 	return allErrs
 }
 
 func ValidateIdentity(identity *api.Identity) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
+	allErrs = append(allErrs, kvalidation.ValidateObjectMeta(&identity.ObjectMeta, false, ValidateIdentityName).Prefix("metadata")...)
+
+	if len(identity.ProviderName) == 0 {
```

Comment:
provider name has restrictions.  Using the expected identity name covers, but does not produce nice messages.